feat: add Docker support for Folksonomy API (#254)
openfoodfacts/folksonomy_api: README.md

Please note that running tests empties the database. DO NOT RUN TESTS in production.

# Docker Setup (Recommended)

> The easiest way to get started with Folksonomy API is to use Docker. This approach requires minimal setup and provides a consistent development environment.

Reviewer: I'm not sure it's the easiest way. It really depends on who use it.

```suggestion
An easy way to get started with Folksonomy API is to use Docker (if you don't mind using it). This approach requires minimal setup and provides a consistent development environment.
```

Reviewer: Indeed I think you should create a local_settings_docker.py which only rely on the environment (which is the right way to do it). I'm not sure we even need this, because I think we already made most settings use environment.

## Requirements
- Docker

> - Docker Compose

Reviewer: Give a minimal version number, as we sometimes have problem with that.

Author: Version Updated in Readme.md

## Quick Start

1. Clone the repository
   ```bash
   git clone https://github.com/openfoodfacts/folksonomy_api.git
   cd folksonomy_api
   ```

2. Start the services
   ```bash
   docker-compose up -d

>    ```

Reviewer: Use `docker compose` instead of `docker-compose` (deprecated)

consistently which my comment above, add the migration run (`docker compose run --rm folksonomy_api python db-migration.py`, if we set postgres as a dependency of folksonomy_api)

3. Access the API
   - API: http://localhost:8000
   - Interactive documentation: http://localhost:8000/docs

4. Stop the services
   ```bash
   docker-compose down
   ```

## Configuration

The Docker setup uses environment variables defined in the `docker-compose.yml` file. You can modify these as needed:

- `POSTGRES_USER`: Database username
- `POSTGRES_PASSWORD`: Database password  
- `POSTGRES_DATABASE`: Database name
- `POSTGRES_HOST`: Database host (default: db)

## Database Access

If you need to access the PostgreSQL database directly, you can use:

```bash

> psql -h localhost -p 5433 -U folksonomy -d folksonomy

Reviewer: why rely on having psql localy instead of using the container ?

```bash
docker compose exec -u postgres folksonomy_db psql
```

Author: corrected with :
docker compose exec db psql -U folksonomy -d folksonomy
